riscv/src/code_gen.rs

```diff
@@ -319,43 +375,154 @@ fn preamble<T: FieldElement>(runtime: &Runtime, with_bootloader: bool) -> String
         + &bootloader_preamble_if_included
         + &memory(with_bootloader)
         + r#"
-    // ============== Constraint on x0 =======================
-
-    x0 = 0;
-
     // ============== iszero check for X =======================
     let XIsZero = std::utils::is_zero(X);
 
+    // =============== Register memory =======================
+    std::machines::memory::Memory regs;
+    instr get_reg Y -> X link ~> X = regs.mload(Y, STEP);
+    instr set_reg X, Y -> link ~> regs.mstore(X, STEP, Y);
+    reg val1;
+    reg val2;
+    col witness val1_col;
+    col witness val2_col;
+    col witness val3_col;
+    col witness val4_col;
+
+    col witness XX, XX_inv, XXIsZero;
+    std::utils::force_bool(XXIsZero);
+    XXIsZero * XX = 0;
+
+    // HACK: This constraint cannot be active globally, because when
+    // XX is not constrained, witgen will try to set XX, XX_inv and XXIsZero
```

Comment:
could also use a hint

Reply:
ok let me check that

Reply:
language wise it would work but witgen doesn't work atm with this because of the order of how the identities and hints are handled